### ISSUE TITLE
Please make it easier to add a node as the root for a scene that already has nodes in it

### ISSUE THREAD

**Issue by alexzheng** [godotengine/godot#9363]
**Operating system or device - Godot version:**
ALL

**Issue description:**

it would be embarrassed that sometimes, I have already created a scene, and I just want to modify the scene to add a new node as the root and make the existed nodes as the child of the new root .(I did not mean to change the type of the root node), but it seems it's impossible to achieve this without creating a new scene to replace the current one.

**Steps to reproduce:**

**Link to minimal example project:**

**Comment by Nodragem** [comment]
I had the same problem, and my solution was to modified the .tscn file within a text editor. 
Would be great to be able to do it simply in the GUI.

**Comment by reduz** [comment]
Could add a rmb option for this

**Comment by bojidar-bg** [comment]
Related #337, #3723 (probably even duplicate of that latter).

**Comment by eon-s** [comment]
Another option is to make a new scene and merge the bad one into it.

The rmb list is already too bloated on the tree, if this is added could be on the scene menu instead?

**Comment by TeddyDD** [comment]
Why not use the reparent dialog? When user try to reparent root to other node just swap them.

**Comment by Nodragem** [comment]
Personally, when that happened to me, the first thing I tried to do is:
- create a Node2D,
- try to drag and drop my old branch (set of nodes) under the newly created Node2D.

Obviously, that did not work, and I was very surprised.
But my point is that this method seemed pretty natural to me, and it does not require the use of any menu.

**Comment by TX256** [comment]
Well, in my mind the simplest and most intuitive way to do this is by dragging any node to the very top of the scene-area (so that the first red line appears). And, then possibly show a confirmation dialog (with possible extra options) for making the dragged node the root?

**Comment by Noshyaar** [comment]
I want to implement the "add node in place" function. The shortcut could be holding shift when clicking new node button. When no node or root node is selected, the editor would behave like the author described. When selecting node in tree, it'd add the selected nodes as child.

E.g. when adding Node2D in place
```
RigidBody2D
    [Sprite] (selected)
    CollisionShape2D
```
would become 
```
RigidBody2D
    Node2D
        Sprite
    CollisionShape2D
```
I'm not sure how it should behave when nodes of multiple levels are selected.

**Comment by rosshadden** [comment]
One of the things I had tried (after dragging/dropping and looking in the RMB menu) was using the reparent dialog to set the intended root node's parent to itself, and another was to reparent the current root node to the intended one.  The latter of course wouldn't be able to work because the intended root node could have siblings.

I don't think "swap with" is general enough, or even useful enough.  Often users want to add a parent to the current root, and  swapping would actually be more work.  Consider this example:

```
root
  someChild
  newRoot
```

Swapping `root` and `newRoot` would make `someChild` now be a child of `newRoot`, when the default expected functionality would be nesting both `root` and `someChild` under `root`.

On a similar note, I thought of "make root node", but it's way too specific.  I like the dragging stuff people mentioned, and I'm sure that's what we all tried first.  But also I think we can come up with a sane way to make the reparent dialog handle this case.  Since that's, you know, it's purpose!

**Comment by ellogwen** [comment]
Any new information according this problem? I still don't know how I can add a new node as root to an existing scene.

**Comment by garyo** [comment]
I keep needing this too -- I'm a newb, but there will always be newbs like me...

**Comment by NathanWarden** [comment]
Just a thought, but wouldn't a "Copy/Paste Node Properties" feature (that will almost guaranteed be asked for at some point) achieve the same thing? You could copy the properties from the node you wanted to be the root and paste them to the node that is currently the root, then re-parent whatever is under the node you copied to the one you pasted to.

**Comment by eon-s** [comment]
The multiple options to operate with nodes in the tree should be documented for new users (merge, copy properties, select and move multiple nodes). 

Other engines do not let you touch the root (like, the Scene or Actor roots) and changing a root is not even a common tree operation, I think something like this, if implemented, should be discouraged.

**Comment by hubbyist** [comment]
Godot scene structure has no distinction as "root only nodes" so in theory all nodes can be root so a right mouse button option as suggested by @reduz is the most suitable way of achieving this for godot. UX considerations must be least concern since there are already duplicates on this issue.

**Comment by bojidar-bg** [comment]
@NathanWarden Note that this is the way it is currently done. It is not very useable unfortunately...

**Comment by NathanWarden** [comment]
@bojidar-bg I'm not sure how that's the way it's currently done. I don't see anything that allows copying all of a node's properties and applying them to another one, unless I'm missing something?

Picture having a root node of type "Node" and a node under it with a type "Spatial". You could copy the Spatial node properties and paste it to the "Node" (or vice versa) and it will become "Spatial" with all the properties/resources/script that the "Spatial" you copied from.

It's maybe not the best way to get this particular problem solved, but would definitely be a more generalized way to solve a slew of problems with wrong node setups including this one.

**Comment by vnen** [comment]
@NathanWarden 

![godot_2018-02-13_13-44-30](https://user-images.githubusercontent.com/5599796/36158740-2cefa8ae-10c4-11e8-8e80-595cade95612.png)

If you change the root node type and copy/paste the params from another node, it's the same as replacing the root. But it's quite a convoluted process.

**Comment by NathanWarden** [comment]
@vnen Ahhh, thanks. Yeah it is a little convoluted as that makes it like 7 or 8 steps to achieve :/

**Comment by brylie** [comment]
Dragging and dropping a node from child position to root position would be simpler and more natural (like shuffling cards). That way, all properties and any scripts stay attached.

**Comment by canadaduane** [comment]
As a newcomer I just encountered this, and thought a use case might be valuable:

I had created my first re-usable scene, and made a `Sprite` the root node. Later, I realized it needed to be animated, so I changed its type to `AnimatedSprite`. Finally, I realized I needed to detect collisions with the player, and wanted to insert a `KinematicBody2D` at the root, with the `AnimatedSprite` as one of its children.

I think this is a fairly natural progression for game devs: objects can initially represent "TODOs" that the dev intends to implement in the future. The ability to add functionality in scenes seems like a good way to evolve placeholders into full-featured objects that interact with the rest of the game.

The ability to re-parent a root node is just one necessary ability in a handful of features that can enable progressive game feature development.

**Comment by alexzheng** [comment]
Maybe a general Add Parent Node menu for any node would be straightforward and useful.

**Comment by AlansCodeLog** [comment]
I recently started learning Godot and this was the first problem I ran into. I would prefer a drag/drop solution but I also like the Add Parent Node option, but something like Wrap Selected Nodes would be even better.

If the dragging isn't implemented though I feel like at the very least you shouldn't get that line that indicates where you're hovering to when you do it around the root node because it makes it seem like you can when you can't. It should be red or non-existent, or give a warning on drop or something.

**Comment by keyframer** [comment]
I can't believe how clumsy it is to change your mind about what the root node should be.  Just let me drag a child above the top node and reparent everything under it.

**Comment by kelvin13** [comment]
this was one of the first questions i had learning godot

**Comment by mhilbrunner** [comment]
As stated in #19035 there is now an option to make a child node the new root node.

**Comment by xo6a** [comment]
Now you can use context menu on new parent node, and select "Make Scene Root"

**Comment by nathancarter** [comment]
For anyone looking for this feature now, I find that:

 1. The fix mentioned by @mhilbrunner in 2018 in #19035 does not seem to still be available in 2024.
 2. The current method seems to be this: first add into the scene somewhere (anywhere) the new node you want to be the new root, then right-click that node, and choose "Make Scene Root."

This method is not intuitive.  I had to google, find and read this issue to the very end, follow the link to the commit with a suggested fix, try it and find that it no longer works, come back here and read @xo6a 's very brief comment that hints at the right method, then go experiment it to see if I understood it correctly.  I'd highly recommend some improvement to this situation, such as a RMB option on the existing root (where people seem to look) that would do one of the highly-upvoted suggestions earlier in this thread.
